nilearn/input_data/fmriprep_confounds.py

```diff
@@ -0,0 +1,306 @@
+          and DVARS.
+        - "full" also remove continuous segments containing fewer than 5
+          volumes.
+
+        One-hot encoding vectors are added as regressors for each scrubbed
+        frame.
+
+    compcor : {'anat_combined', 'anat_separated', 'temporal',
+    'temporal_anat_combined', 'temporal_anat_separated'}
+
+        .. warning::
+            Require fmriprep >= v:1.4.0.
+
+        Type of confounds extracted from a component based noise correction
+        method.
+
+        - "anat_combined" noise components calculated using a white matter and
+          CSF combined anatomical mask
+        - "anat_separated" noise components calculated using white matter mask
+          and CSF mask compcor separately; two set of scores are concatenated
+        - "temporal" noise components calculated using temporal compcor
+        - "temporal_anat_combined" components of "temporal" and "anat_combined"
+        - "temporal_anat_separated" components of "temporal" and
+          "anat_separated"
+
+    n_compcor : "all" or int, default "all"
+        The number of noise components to be extracted.
+        For acompcor_combined=False, and/or compcor="full", this is the number
+        of components per mask.
+        "all": select all components (50% variance explained by fMRIPrep
+        defaults)
+
+    ica_aroma : {'full', 'basic'}
+
+        - "full": use fMRIprep output `~desc-smoothAROMAnonaggr_bold.nii.gz`.
+        - "basic": use noise independent components only.
+
+    demean : boolean, default True
+        If True, the confounds are standardized to a zero mean (over time).
+        This step is critical if the confounds are regressed out of timeseries
+        using nilearn.signal.clean with `standardize=False` or
+        `standardize='zscore'`, but should be turned off with
+        `standardize='psc'`.
+
+
+    Returns
+    -------
+    confounds : pandas.DataFrame, or list of
+        A reduced version of fMRIprep confounds based on selected strategy
+        and flags.
+        An intercept is automatically added to the list of confounds.
+        The columns contains the labels of the regressors.
+
+    sample_mask : None, numpy.ndarray, or list of
+        When no volumns require removal, the value is None.
+        Otherwise, shape: (number of scans - number of volumes removed, )
+        The index of the niimgs along time/fourth dimension for valid volumes
+        for subsequent analysis.
+        This attribute should be passed to parameter `sample_mask` of
+        :class:`nilearn.input_data.NiftiMasker` or
+        :func:`nilearn.signal.clean`.
+        Volumns are removed if flagged as following:
+
+        - Non-steady-state volumes (if present)
+        - Motion outliers detected by scrubbing
+
+    Notes
+    -----
+    The noise components implemented in this class are
+    adapted from (Ciric et al. 2017). Band-pass filter is replaced
```

Comment:
Since there are several scientific references, we could improve the docstring by adding the BibTex refs to `nilearn/doc/references.bib`.

You would then only need to replace this line by:

```
adapted from :footcite:`Ciric2017`. Band-pass filter is replaced
```

(assuming the bibtex entry is `Ciric2017`).

And the References section will simply become:

```
References
-----------
.. footbibliography::
```

(nilearn/datasets/atlas.py is formatted this way if you want an example).

The references will then be much easier to maintain, always formatted properly, and appear in the main Nilearn's [bibliography](https://nilearn.github.io/bibliography.html).
WDYT?

Reply:
Thanks for providing the instructions!